resources/aws/security_group.go

```diff
@@ -61,6 +62,94 @@ func (s SecurityGroup) findExisting() (*ec2.SecurityGroup, error) {
 	return securityGroups.SecurityGroups[0], nil
 }
 
+// findGroupWithRule checks if a security group exists with the specified name,
+// description and rule. SourceCIDR always takes precedence over SecurityGroupID.
+func (s SecurityGroup) findGroupWithRule(rule SecurityGroupRule) (*ec2.SecurityGroup, error) {
+	var filters []*ec2.Filter
+	if rule.SourceCIDR != "" {
+		filters = []*ec2.Filter{
+			&ec2.Filter{
+				Name: aws.String(subnetDescription),
+				Values: []*string{
+					aws.String(s.Description),
+				},
+			},
+			&ec2.Filter{
+				Name: aws.String(subnetGroupName),
+				Values: []*string{
+					aws.String(s.GroupName),
+				},
+			},
+			&ec2.Filter{
+				Name: aws.String(ipPermissionCIDR),
+				Values: []*string{
+					aws.String(rule.SourceCIDR),
+				},
+			},
+			&ec2.Filter{
+				Name: aws.String(ipPermissionFromPort),
+				Values: []*string{
+					aws.String(strconv.Itoa(rule.Port)),
+				},
+			},
+			&ec2.Filter{
+				Name: aws.String(ipPermissionProtocol),
+				Values: []*string{
+					aws.String(rule.Protocol),
+				},
+			},
+			&ec2.Filter{
+				Name: aws.String(ipPermissionToPort),
+				Values: []*string{
+					aws.String(strconv.Itoa(rule.Port)),
+				},
+			},
+		}
+	} else {
+		filters = []*ec2.Filter{
```

Comment:
Can you elaborate? Is this AWS specific or the operator specific?

Can we have common subset of filters and append `SourceCIDR` or `SecurityGroupID` in the `if`s?